tests/test_helpers.py

```diff
@@ -515,6 +515,71 @@ def test_proxies_from_env_http_with_auth(mocker) -> None:
     assert proxy_auth.encoding == "latin1"
 
 
+# --------------------- get_env_proxy_for_url ------------------------------
+
+
+def test_get_env_proxy_for_url_when_url_matches_the_no_proxy_list(mocker) -> None:
+    mocker.patch("aiohttp.helpers.proxy_bypass", return_value=True)
+    mocker.patch(
+        "aiohttp.helpers.proxies_from_env",
+        return_value={"http": helpers.ProxyInfo("http://example.com", "")},
+    )
+    url = URL("http://aiohttp.io/path")
+    with pytest.raises(LookupError, match=r"Proxying is disallowed for `'aiohttp.io'`"):
+        helpers.get_env_proxy_for_url(url)
+
+
+def test_get_env_proxy_for_url_when_url_scheme_does_not_match_proxy_list(
+    mocker,
+) -> None:
+    mocker.patch("aiohttp.helpers.proxy_bypass", return_value=False)
+    proxies = {"https": helpers.ProxyInfo("https://example.com", "")}
+    mocker.patch("aiohttp.helpers.proxies_from_env", return_value=proxies)
+    url = URL("http://aiohttp.io/path")
+
+    with pytest.raises(
+        LookupError, match=r"No proxies found for `http://aiohttp.io/path` in the env"
+    ):
+        helpers.get_env_proxy_for_url(url)
+
+
+def test_get_env_proxy_for_url_when_url_is_empty(
+    mocker,
+) -> None:
+    mocker.patch("aiohttp.helpers.proxy_bypass", return_value=False)
+    proxies = {"https": helpers.ProxyInfo("https://example.com", "")}
+    mocker.patch("aiohttp.helpers.proxies_from_env", return_value=proxies)
+    url = URL("")
+
+    with pytest.raises(LookupError, match=r"No proxies found for `` in the env"):
+        helpers.get_env_proxy_for_url(url)
+
+
+@pytest.mark.parametrize(
+    "schema,proxyInfo,url",
```

Comment:
Apparently, pytest lets you specify the params separately for readability:
```suggestion
    ("schema", "proxyInfo", "url"),
```